extensions/amp-story-shopping/0.1/amp-story-shopping-tag.js

```diff
@@ -85,10 +93,18 @@ export class AmpStoryShoppingTag extends AMP.BaseElement {
     });
   }
 
-  /**
-   * @private
-   */
+  /** @private */
+  setPageAttachmentImpl_() {
+    this.element
+      .closest('amp-story-page')
+      .querySelector('amp-story-page-attachment')
+      .getImpl()
+      .then((impl) => (this.pageAttachmentImpl_ = impl));
+  }
+
+  /** @private */
   onClick_() {
+    this.pageAttachmentImpl_.open();
```

Comment:
Is there any other way to open the attachment that doesn't involve 2 query selectors + a hard reference on a component instance?
If it's too complicated, I'd be OK with just not storing the attachmentImpl_ in an ivar and retrieving it each time.

Also just curious, how will you know what product has been clicked and should be displayed?

Reply:
> Is there any other way to open the attachment that doesn't involve 2 query selectors + a hard reference on a component instance?

If there is I'm all ears. Every other way I can think of still requires a hard reference and code outside of this component. What is the benefit retrieving it each time over keeping a hard reference?

> Also just curious, how will you know what product has been clicked and should be displayed?

The active product is set in this onClick function.
The store will be queried from `amp-story-shopping-attachment` when the page attachment opens.
[WIP PR](https://github.com/ampproject/amphtml/pull/37535/files#diff-3ef47b0cbc87c60235a54da08fc86c452af88994f1ba184e38be8d003e7b2552R81-R99)  if interested in how it's looking.

Reply:
The attachment could listen and open/close based on attachment state updates.
Then buttons would set attachment state rather than calling open() directly.
We would also need to pass `shouldAnimate` through that state.
What do you think? It could be a refactor in a separate PR but would block this.